nrf_modem/doc/sockets.rst

```diff
@@ -0,0 +1,371 @@
+.. _nrf_sockets:
+
+Networking sockets
+##################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+The Modem library offers an implementation of a subset of the POSIX networking sockets API.
+This API allows you to reuse existing programming experience and port existing simple networking applications to the nRF9160.
+The implementation is namespaced (``nrf_``) to avoid conflicts with functions that might be part of libc or other POSIX compatibility libraries.
+
+Following is a list of supported functions:
+
+* :c:func:`nrf_socket`
+* :c:func:`nrf_close`
+* :c:func:`nrf_connect`
+* :c:func:`nrf_bind`
+* :c:func:`nrf_listen`
+* :c:func:`nrf_send`
+* :c:func:`nrf_sendto`
+* :c:func:`nrf_recv`
+* :c:func:`nrf_recvfrom`
+* :c:func:`nrf_setsockopt`
+* :c:func:`nrf_getsockopt`
+* :c:func:`nrf_poll`
+* :c:func:`nrf_getaddrinfo`
+* :c:func:`nrf_freeaddrinfo`
+* :c:func:`nrf_getifaddrs`
+* :c:func:`nrf_freeifaddrs`
```

Comment:
These two are actually not there yet, so we should remove them from this list for the time being.